btf_types/src/lib.rs

```diff
@@ -0,0 +1,196 @@
+            '+' => Ok(RawInstructions::IncrementByte),
+            '-' => Ok(RawInstructions::DecrementByte),
+            '.' => Ok(RawInstructions::OutputByte),
+            ',' => Ok(RawInstructions::AcceptByte),
+            '[' => Ok(RawInstructions::ZeroJump),
+            ']' => Ok(RawInstructions::NonZeroJump),
+            _ => Err("Non BF instruction"),
+        }
+    }
+}
+
+/// Provide human-readable format of the instructions.
+impl fmt::Display for RawInstructions {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            RawInstructions::IncrementDataPointer => write!(f, "Increment data pointer"),
+            RawInstructions::DecrementDataPointer => write!(f, "Decrement data pointer"),
+            RawInstructions::IncrementByte => write!(f, "Increment byte"),
+            RawInstructions::DecrementByte => write!(f, "Decrement byte"),
+            RawInstructions::OutputByte => write!(f, "Output byte"),
+            RawInstructions::AcceptByte => write!(f, "Accept byte"),
+            RawInstructions::ZeroJump => write!(f, "Zero jump"),
+            RawInstructions::NonZeroJump => write!(f, "Non zero jump"),
+        }
+    }
+}
+
+/// Provide structure to reprsent location of BF instruction in file.
+#[derive(Debug)]
+pub struct IntructionPosition {
+    /// BF instruction.
+    instruction: RawInstructions,
+    /// Line of the file from where BF instruction is parsed.
+    line: usize,
+    /// Positoin at the line from where BF instruction is parsed.
+    position: usize,
+}
+
+impl IntructionPosition {
+    /// Get parsed instruction.
+    pub fn instruction(&self) -> &RawInstructions {
+        &self.instruction
+    }
+
+    /// Get line of parsed instruction.
+    pub fn line(&self) -> usize {
+        self.line
+    }
+
+    /// Get position at the line of parsed instruction.
+    pub fn position(&self) -> usize {
+        self.position
+    }
+}
+
+/// Provide human-readable format of the instruction with position in parsed file.
+impl fmt::Display for IntructionPosition {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "{}:{}] {}", self.line, self.position, self.instruction)
+    }
+}
+
+/// Provide structure to reprsent BF program.
+#[derive(Debug)]
+pub struct BrainFuckProgram {
+    /// Name of the file from where program is parsed.
+    filename: String,
+    /// List of instructions with location parsed from file.
+    instructions: Vec<IntructionPosition>,
+}
+
+impl BrainFuckProgram {
+    /// Create BF program based on the name of the file and it's content.
+    fn new(filename: String, content: String) -> Self {
+        let mut instructions: Vec<IntructionPosition> = Vec::new();
+
+        let mut line: usize = 1;
+        let mut position: usize = 1;
+        for char in content.chars() {
+            if char == '\n' {
+                line += 1;
+                position = 0;
+            }
```

Comment:
Should there be a `continue` in this `if` block, or are you deliberately falling through?

Reply:
I am failing through for a purpose